src/Components/Config/configHook.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+import { configEndpoint, header } from '../../apiCalls';
+
+import { ApiConfig } from '../../Types/ApiFormData';
+
+async function getConfig() {
+  // fetch data
+  return fetch(configEndpoint, {
+    method: 'GET',
+    headers: header,
+  }).then((response) => {
+    if (!response.ok) {
+      throw new Error(`${response.status} ${response.statusText}`);
+    }
+    return response.json();
+  });
+}
+
+export default function useGetConfig() {
+  const [loading, setLoading] = useState<boolean>(true);
+  const [config, setConfig] = useState<ApiConfig | undefined>();
+
+  useEffect(() => {
+    getConfig().then((value) => {
+      // get data and set loading to false
```

Comment:
Can you loop through the api response, and create an object where the key is the name, and the value is the data.

Reply:
Would accessing the data as the example below not give us the needed object?

`const configData = Array.isArray(value) && value[0]?.data;`

Reply:
This requires all of the config to be in one model on the backend. The config response should look like:

```
[
  {
    "name": "zipcode",
    "data": [80205, zipcodes...]
  },
  {
    "name": "other",
    "data": {"other": "data"}
  }
]
```
and then be converted to:
```
{
  "zipcode": [80205, zipcodes...],
  "other": {"other": "data"}
}
```